chapter_preliminaries/probability.md

```diff
@@ -354,54 +808,235 @@ $$P(H = 1| D_1 = 1, D_2 = 1)
+has coordinates $\mu_i = E_{\mathbf{x} \sim P}[x_i]$.
+Covariances are more complicated. 
+We resolve the problem by taking expectations of the *outer product* 
+of the difference between random variables and their mean. 
+
+$$\Sigma := \mathrm{Cov}_{\mathbf{x} \sim P}[\mathbf{x}] = E_{\mathbf{x} \sim P}\left[(\mathbf{x} - \mathbf{\mu}) (\mathbf{x} - \mathbf{\mu})^\top\right].$$
+
+This matrix $\Sigma$ is referred to as the covariance matrix. 
+An easy way to see its effect is to consider some vector $\mathbf{v}$ 
+of the same size as $\mathbf{x}$. 
+It follows that 
+
+$$\mathbf{v}^\top \Sigma \mathbf{v} = E_{\mathbf{x} \sim P}\left[\mathbf{v}^\top(\mathbf{x} - \mathbf{\mu}) (\mathbf{x} - \mathbf{\mu})^\top \mathbf{v}\right] = \mathrm{Var}_{x \sim P}[\mathbf{v}^\top \mathbf{x}].$$
+
+As such, $\Sigma$ allows us to compute the variance 
+for any linear function of $\mathbf{x}$
+by a simple matrix multiplication. 
+The off-diagonal elements tell us how correlated coordinates are:
+a value of 0 means no correlation, 
+whereas a large positive value 
+means that they are strongly correlated. 
+
+
+
+## Discussion
+
+In machine learning, there are many things to be uncertain about!
+We can be uncertain about the value of a label given an input.
+We can be uncertain about the estimated value of a parameter.
+We can even be uncertain about whether data arriving at deployment
+is even from the same distribution as the training data.
+
+By *aleatoric uncertainty*, we denote that uncertainty 
+that is intrinsic to the problem, 
+and due to genuine randomness 
+unaccounted for by the observed variables, 
+By *epistemic uncertainty*, we denote uncertainty
+over a model's parameters, the sort of uncertainty
+that we can hope to reduce by collecting more data.
+We might have epistemic uncertainty 
+concerning the probability 
+that a coin turns up heads,
+but even once we know this probability,
+we are left with aleatoric uncertainty 
+about the outcome of any future toss.
+No matter how long we watch someone tossing a fair coin, 
+we will never be more or less than 50% certain 
+that the next toss will come up heads.
+These terms owe to literature in mechanical modeling,
+(see e.g., :cite:`der2009aleatory` for a review on this aspect of [Uncertainty Quantification](https://en.wikipedia.org/wiki/Uncertainty_quantification)).
+It's worth noting that these terms constitute a slight abuse of language.
+The term *epistemic* refers to anything concerning *knowledge*
+and thus in the philosophical sense, all uncertainty is epistemic.
+
+
+We saw that sampling data from some unknown probability distribution 
+can provide us with information that can be used to estimate
+the parameters of the data generating distribution.
+That said, the rate at which this is possible can be quite slow. 
+In our coin tossing example (and many others) 
+we can do no better than to design estimators
+that converge at a rate of $1/\sqrt{n}$. 
```

Comment:
Undefined $n$

Reply:
Sure, but the definition is a bit far away so readers (like me) may have to stop here and think "what is n" (breaking the flow) before moving on. Shall we recall what $n$ is here?